nncf/experimental/post_training/algorithms/quantization/min_max/onnx_backend.py

```diff
@@ -0,0 +1,117 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from copy import deepcopy
+import numpy as np
+import onnx
+
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.common.tensor_statistics.collectors import ReductionShape
+from nncf.common.utils.backend import BackendType
+from nncf.common.utils.registry import Registry
+
+from nncf.experimental.onnx.graph.metatypes.onnx_metatypes import GENERAL_WEIGHT_LAYER_METATYPES
+from nncf.experimental.onnx.graph.model_transformer import ONNXModelTransformer
+from nncf.experimental.onnx.graph.transformations.commands import ONNXQuantizerInsertionCommand
+from nncf.experimental.onnx.graph.transformations.commands import ONNXTargetPoint
+from nncf.experimental.onnx.hardware.config import ONNXHWConfig
+from nncf.experimental.onnx.hardware.fused_patterns import ONNX_HW_FUSED_PATTERNS
+from nncf.experimental.onnx.quantization.default_quantization import DEFAULT_ONNX_QUANT_TRAIT_TO_OP_DICT
+from nncf.experimental.onnx.statistics.collectors import ONNXMeanMinMaxStatisticCollector
+from nncf.experimental.onnx.statistics.collectors import ONNXMinMaxStatisticCollector
+
+from nncf.experimental.post_training.algorithms.quantization.min_max.backend import MinMaxAlgoBackend
+from nncf.experimental.post_training.algorithms.quantization.min_max.backend import ALGO_BACKENDS
+from nncf.experimental.post_training.algorithms.quantization.min_max.utils import QuantizerLayerParameters
+
+
+@ALGO_BACKENDS.register(BackendType.ONNX)
+class ONNXMinMaxAlgoBackend(MinMaxAlgoBackend):
```

Comment:
Some return type hints seem to be missing there. Should be safe to set up even the backend-specific return types here.

Reply:
Thank you for your comment! I really missed a number of hints. Fixed, please, review it.